src/Compilers/VisualBasic/Portable/Symbols/SymbolExtensions.vb

```diff
@@ -380,14 +380,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
         <Extension>
         Friend Function GetDeclaringSyntaxNode(Of T As VisualBasicSyntaxNode)(this As Symbol) As T
 
-            For Each node In this.DeclaringSyntaxReferences.Select(Function(d) d.GetSyntax())
-                Dim node_T = TryCast(node, T)
+            For Each reference In this.DeclaringSyntaxReferences
+                Dim node_T = TryCast(reference.GetSyntax(), T)
                 If node_T IsNot Nothing Then
                     Return node_T
                 End If
             Next
 
-            Debug.Assert(this.IsImplicitlyDeclared)
+            Debug.Assert(this.IsImplicitlyDeclared OrElse (TypeOf this Is SourceMemberContainerTypeSymbol And
```

Comment:
That's what I had initially, but I ran into problems with option Strict On.
`TryCast(this, SourceMemberContainerTypeSymbol)?.SyntaxReferences.All(Function(ref) ref.SyntaxTree.IsEmbeddedOrMyTemplateTree()) = True)` // Option Strict On disallows implicit conversions from 'Boolean?' to 'Boolean'.

Also, I think that logic was incorrect anyways (in the case when `this` is not a `SourceMemberContainerTypeSymbol` the assert should succeed).

I will fix the `And` -> `AndAlso` problem though.